### PR TITLE
Limit the list of campaigns visible to an user to the org units that are under his responsability

### PR DESCRIPTION
**Pull request by xavierbarbosa**
### Updated
- Ony show campaigns for a user's org unit or a descendant of the org unit that is assigned in his profile

---
Closes https://bluesquare.atlassian.net/browse/IA-569

### CODE REVIEW

File: plugins/polio/api.py [hunk 1]
```diff
@@ -28,6 +30,17 @@ def list(self, request: Request, *args, **kwargs):
         else:
             return Response(serializer.data)
 
+    def get_queryset(self):
+        user = self.request.user
+
+        if user.iaso_profile.org_units.count():
+            org_units = map(lambda org_unit: OrgUnit.objects.hierarchy(org_unit), user.iaso_profile.org_units.all())
+            org_units = list(itertools.chain.from_iterable(org_units))
```
Comment (olethanh) [inline plugins/polio/api.py:38]: hierarchy() can already take a list  or queryset of orgunit so this shouldn't be needed

Comment (xavierbarbosa) [reply plugins/polio/api.py]: Nice catch, I missed that for some reason.

**Comment by olethanh** [comment]
re reading. Shouldn't we also filter for rights on OrgUnit (via account and projects)? Martin?

**Comment by xavierbarbosa** [comment]
@olethanh I retrieve the org units from the user account, but I don't get what this means for Projects.

**Comment by madewulf** [comment]
Olivier is right that this feature does not take into account multitenancy of Iaso, but then, we should add an account field to campaigns, but on the other hand, I don't see this being used as multi tenant.

**Pull request merged**